lib/core/sync_job_runner.rb

```diff
@@ -83,10 +85,12 @@ def do_sync!
         Utility::Logger.info("Upserted #{@status[:indexed_document_count]} documents into #{@connector_settings.index_name}.")
         Utility::Logger.info("Deleted #{@status[:deleted_document_count]} documents into #{@connector_settings.index_name}.")
 
+        @status[:error] = 'Sync thread didn\'t finish execution. Check connector logs for more details.' unless @sync_finished
```

Comment:
I think also this line is worth the comment that explains why is has to be done this way - about the interrupts and how they are handled in a multithreaded app